### ISSUE TITLE
Allow oxygen to be obtained from SiO2 in 2.1.2.0

### ISSUE THREAD

**Issue by OrderedSet86** [GTNewHorizons/GT-New-Horizons-Modpack#9072]
### Your GTNH Discord Username

Order#0001

### Your Pack Version

2.1.2.0-pre5

### Your Proposal

SiO2 used to be able to be electrolyzed. This was removed in probably one of these PRs:
- https://github.com/GTNewHorizons/GT5-Unofficial/commit/473d775e0e781656fecba4183066885b45ebf081
- https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/8623#issuecomment-991445324

This breaks the last step of the cobble -> oxygen line available in MV:
Cobble (LV rock breaker)
Gravel (LV forge hammer)
Sand (LV forge hammer)
Glass (MV arc furnace)
Glass Dust (LV forge hammer)
Silicon Dioxide (LV centrifuge)
Silicon + Oxygen (LV electrolyzer)

To keep the old line, I would like to be able to still get oxygen from SiO2, regardless of whether that's a newly designed route or revert for SiO2 electrolyzing.

From what I could tell from Discord #dev-offtopic discussion, the intent was to make electrolyzing less of a "do everything machine." However, this change does not move things away from the electrolyzer - the alternative oxygen sources mostly still use the electrolyzer (sugar, water, ore dusts).

### Your Goal

Either add another way to get oxy from SiO2 or revert change. Suggestions below.

[Looking at how O2 might be obtained from silica (SiO2) IRL](https://chemistry.stackexchange.com/questions/49187/can-you-get-oxygen-form-silica-or-alumina), it looks like it's just heating -> electrolyzing. As a result, here are some other possible lines:
1. SiO2 -> fluid extractor (silica fluid, hot 1983K) -> electrolyzer -> silicon dust + oxygen gas
2. Same as 1, but silicon dust is hot and must be cooled down with coolant or vac freezer
3. Same as 1 or 2, but with a fluid heater step before electrolyzing (although this would make less chemical sense that it would be a fluid before being hot enough)
4. SiO2 -> EBF -> silicon dust + oxygen gas (not sure how this would work with muffler tier)

### Your Vision

This change/revert would continue to allow robust non-crop methods to obtain oxygen.

In addition, this setup is a great introduction to fully automated processing lines. It is one of the first lines I built in my 2.1.1.0 run so I could get lots of steel on-demand. Picture below. Plus it's a great source of glass for base decoration.

![2021-08-16_22 01 44](https://user-images.githubusercontent.com/88752681/145730780-74641086-dc34-4d09-925a-21cd9610ffbe.png)

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying *I like this change please reconsider adding it* will prompt us to investigate and reopen it once we confirm your report.
- [X] I understand this change request may not attract enough attention and thus not be implemented.
- [X] I understand this change request may be rejected due to other community members think it's inappropriate.
- [X] I believe this feature would make the pack better.

**Comment by OrderedSet86** [comment]
Another option (suggested by @Glease - maybe jokingly) if the goal is to nerf silicon: just remove the silicon output from the electrolyzer recipe. There is already a route to get silicon from putting the SiO2 with carbon into the EBF.
![image](https://user-images.githubusercontent.com/88752681/145732746-7122d110-8fec-42c0-bb51-fffd09e9580f.png)

This wouldn't be realistic but it would fix the silicon balancing issue.

**Comment by bombcar** [comment]
Removing the silicon would be fine by me - we need more lossy recipes anyway.

**Comment by Sphyix** [comment]
instead of removing it, just reduce the amount of it

**Comment by bombcar** [comment]
Percent chance of silicon would be fine - too bad we can’t increase percent chance based on overclock

**Comment by bombcar** [comment]
<img width="320" alt="image" src="https://user-images.githubusercontent.com/533931/145735103-1a0fc7ad-b3fe-4501-9305-17ec3bf72cbf.png">

**Comment by Sphyix** [comment]
> <img alt="image" width="320" src="https://user-images.githubusercontent.com/533931/145735103-1a0fc7ad-b3fe-4501-9305-17ec3bf72cbf.png">

no, for the reasons I explained in the ticket you closed, if you even took the time to read it

**Comment by Sphyix** [comment]
Since you closed it, i'll copy paste here

Revert the SiO2 removal.
Reasons:
If the plan was to reduce the use of electrolyzer like someone said on the discord channel, this change did nothing as oxygen still needs to be obtained by electrolyzing dusts and silicon needs to be obtained by electrolyzing dusts, but in smaller quantities.

If the plan was to reduce the amount of silicon, this change did nothing aswell as there's an EBF recipe for silicon ingots using 1 silicon dioxide and 2 carbon, giving 1 silicon ingot.
There's a big issue with this alternative recipe, as this is the easiest way to get silicon, but in the ingot form.
A lot of recipes require silicon dust for the alloys, or silicon dust to make silicon boules in an EBF
This means smelting silicon -> macerating it -> smelting it again, that looks pretty bad.

If the plan was to reduce the amount of oxygen at lower tiers, why not make the process more expensive to run and reduce by 2x or 4x the amount of oxygen?

Another issue I'm personally facing now, I use a TON of silicon dust for the octuple solars, Colen mentioned something around the 16 million silicon, and a good part of it is for wafers, so that means smelting and macerating, let's say, 10 million silicon.
This adds an unnecessary load to the server, as it adds 2 more steps to get silicon dust from silicon dioxide.

At the end, I don't see why this change was made.
Please revert it or explain better why this removal is necessary.

**Comment by OrderedSet86** [comment]
Just to back up @Sphyix 's claim:
1A solar panel = 2 wafers + 2 LV circuits = 64+2 = 66 silicon (using microprocessor recipe for LV circs)
So single compressed solar = 66*8 = 594 silicon
Double compressed solar = 528*8 = 4224 silicon
Triple compressed solar = 33792 silicon
Quadruple compressed solar = 270336 silicon
Quintuple compressed solar = 2.16 million silicon
Sextuple compressed solar = 17.3 million silicon
Septuple compressed solar = 138.4 million silicon
Octuple compressed solar = 1.11 billion silicon

Removing silicon entirely would indeed be a substantial solar nerf. Or maybe solars shouldn't be 8^n recursive...

**Comment by OrderedSet86** [comment]
Just to back up @Sphyix 's claim:
1A solar panel = 2 wafers + 2 LV circuits = 4+2 = 6 silicon (using microprocessor recipe for LV circs)
So single compressed solar = `6*8` = 48 silicon
Double compressed solar = `528*8` = 384 silicon
Triple compressed solar = 3072 silicon
Quadruple compressed solar = 24,576 silicon
Quintuple compressed solar = 196.6k silicon
Sextuple compressed solar = 1.57 million silicon
Septuple compressed solar = 12.6 million silicon
Octuple compressed solar = 100.7 million silicon

Removing silicon from SiO2 would indeed be a solar nerf. There aren't any obvious good late game sources I can see besides SiO2, although maybe void miner ore processing might work.

**Comment by bombcar** [comment]
Freaking recursive crafting needs to die …

**Comment by Savallator** [comment]
I don't get it, silicon is so fucking abundant everywhere... doing this change to nerf silicon looks stupid for me anyways.
For Silicon:
1. plant some Lazullias with a farming station, get tons and also chlorine
2. Redstone
3. Obsidian
4. Ferrosilite
5. All the useless ores (Fullers, Bentonite, Soapstone, Uvavorite, Any Garnet, Mica, Silicon ore, Emerald, Andradite, Grossular, Pyrope, Almandine, Spessartine)

Actually it is hard to find a composite ore without silicon... so i really don't get the discussion about how silicon is so hard, especially lategame...

**Comment by GTNH-Afx237v7** [comment]
But all silicon, especially from the aforementioned „useless“ ores, now comes as silicon dioxide. 
i am also not sure why the effort is necessary to make silicon harder to get. I’d prefer the raw silicon from electrolyzing or at least silicon dioxide dust to silicon dust back. Going through smelting only to macerate again just double the effort (and cpu usage).

**Comment by Sphyix** [comment]
This has been "fixed" already with a centrifuge recipe 3 sio2 -> 1 si + 2 o2 
Energy cost was changed to 10x compared to electrolyzer

**Comment by Savallator** [comment]
yeah, so it was made unusable for early tiers, and the whole chain got useless (later on Oxygen is abundant from other sources).
Also, yeah, Centrifuge... how fitting...

**Comment by Sphyix** [comment]
Read the comments on the pr please :)
I told them, got ignored.
Ez
